### PR TITLE
docs: adds EDR API overview documentation

### PR DESCRIPTION
**Pull request by yurimssilva**
## WHAT

Adds initial documentation for the EDR/Control Plane Adapter API

### CODE REVIEW

**Review by wolf4ood — APPROVED** [review]
LGTM :rocket:

**Review by stephanbcbauer — APPROVED** [review]
LGTM

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    }
+  }
+]
+```
+
+- This endpoint, through the `transfer-process-id` passed as `path variable`, will retrieve the actual EDR.
+
+| Path                                         | Method | Query Params             |
+|----------------------------------------------|--------|--------------------------|
+| `<MANAGEMENT_URL>/edrs/{trasfer-process-id}` | GET    | none                     |
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:112]:
```suggestion
| `<MANAGEMENT_URL>/edrs/{transfer-process-id}` | GET    | none                     |
```

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+> Please note that the `data destination` will always be `HttpProxy`, requiring a request against the provider's `data-plane` to fetch the asset data.
+
+| Path                            | Method | Query Params             |
+|---------------------------------|--------|--------------------------|
+| `<MANAGEMENT_URL>/edrs`         | POST   | none                     |
+
+#### Payload
+
+```json
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:33]: NIT: The formatting of the placeholders seems strange to me, maybe just use `"connectorAddress": "<PROVIDER_DSP_URL>"` instead?

**Review by saschaisele-zf — CHANGES_REQUESTED** [review]

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+| property                                                  | description                                          | required  | default value |
+|-----------------------------------------------------------|------------------------------------------------------|-----------|---------------|
+| edc.datasource.edr.name                                   | Defines the name associated with the EDR data source | false     | edr           |
+| edc.datasource.edr.url                                    | Defines the database address to access the EDR data  | true      |               |
+| edc.datasource.edr.user                                   | Defines EDR database username                        | true      |               |
+| edc.datasource.edr.password                               | Defines EDR database password                        | true      |               |
+| org.eclipse.tractusx.edc.postgresql.migration.edr.enabled | Enables EDR database migration	                      | true      |               |
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:17]: Table formatting.

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+### EDR Management | Fetch cached EDRs
+
+- This endpoint will retrieve all EDR entries by their `assetId` or `agreementId` references, which are passed as `query parameters`.
+
+| Path                                         | Method | Query Params         |
+|----------------------------------------------|--------|----------------------|
+| `<MANAGEMENT_URL>/edrs`                      | GET    | assetId, agreementId |
+
+#### Response
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:87]: Duplicate header is not allowed and will throw a lint error.
```suggestion
#### EDR Entry Response
```

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+]
+```
+
+- This endpoint, through the `transfer-process-id` passed as `path variable`, will retrieve the actual EDR.
+
+| Path                                          | Method | Query Params             |
+|-----------------------------------------------|--------|--------------------------|
+| `<MANAGEMENT_URL>/edrs/{transfer-process-id}` | GET    | none                     |
+
+#### Response
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:114]:
```suggestion
#### EDR Response
```

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    "tx": "https://w3id.org/tractusx/v0.0.1/ns/",
+    "edc": "https://w3id.org/edc/v0.0.1/ns/",
+    "dcat": "https://www.w3.org/ns/dcat/",
+    "odrl": "http://www.w3.org/ns/odrl/2/",
+    "dspace": "https://w3id.org/dspace/v0.8/"
+  }
+}
+```
+
+> Please note that now with the EDR you are able to request the `Asset` data from provider's `data-plane`.
```
Comment (saschaisele-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:136]: No trailing newline.
```suggestion
> Please note that now with the EDR you are able to request the `Asset` data from provider's `data-plane`.

```

**Review by saschaisele-zf — CHANGES_REQUESTED** [review]

File: docs/samples/edr-api-overview/edr-api-overview.md (hunk 4 above)
Comment (florianrusch-zf) [reply docs/samples/edr-api-overview/edr-api-overview.md:87]: We should adapt the above header as well to have some consistency (have a look at my other comment)

File: docs/samples/edr-api-overview/edr-api-overview.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+          "odrl:type": "USE"
+        }
+      },
+      "odrl:target": "<ASSET_ID>"
+    }
+  }
+}
+```
+
+#### Response
```
Comment (florianrusch-zf) [inline docs/samples/edr-api-overview/edr-api-overview.md:61]:
```suggestion
#### EDR Negotiation Response
```

**Review by saschaisele-zf — APPROVED** [review]

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

**Pull request merged**